src/CommunityToolkit.Maui.Core/Platform/StatusBar/StatusBar.android.cs

```diff
@@ -31,35 +30,26 @@ static void PlatformSetStyle(StatusBarStyle style)
 		switch (style)
 		{
 			case StatusBarStyle.DarkContent:
-				AddBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
+				SetStatusBarAppearance(Activity, true);
 				break;
 
 			case StatusBarStyle.Default:
 			case StatusBarStyle.LightContent:
-				RemoveBarAppearanceFlag(Activity, (StatusBarVisibility)SystemUiFlags.LightStatusBar);
+				SetStatusBarAppearance(Activity, false);
 				break;
 
 			default:
 				throw new NotSupportedException($"{nameof(StatusBarStyle)} {style} is not yet supported on Android");
 		}
 	}
 
-	static void AddBarAppearanceFlag(Activity activity, StatusBarVisibility flag) =>
-		SetBarAppearance(activity, barAppearance => barAppearance |= flag);
-
-
-	static void RemoveBarAppearanceFlag(Activity activity, StatusBarVisibility flag) =>
-		SetBarAppearance(activity, barAppearance => barAppearance &= ~flag);
-
-
-	static void SetBarAppearance(Activity activity, Func<StatusBarVisibility, StatusBarVisibility> updateAppearance)
+	static void SetStatusBarAppearance(Activity activity, bool lightStatusBars)
 	{
 		var window = GetCurrentWindow(activity);
-#pragma warning disable CS0618 // Type or member is obsolete
-		var appearance = window.DecorView.SystemUiVisibility;
-		appearance = updateAppearance(appearance);
-		window.DecorView.SystemUiVisibility = appearance;
-#pragma warning restore CS0618 // Type or member is obsolete
+		_ = new WindowInsetsControllerCompat(window, window.DecorView)
+		{
+			AppearanceLightStatusBars = lightStatusBars
+		};
```

Comment:
Is there a reason to not use the existing `WindowInsetsControllerCompat`? If don't, instead of creating a new instance (and discarding it), every time this code executes, we can use the existing one and change the appearance.
```suggestion
		var windowController = WindowCompat.GetInsetsController(window, window.DecorView);
		windowController.AppearanceLightStatusBars = lightStatusBars;
```

I tested it on my end, and that works as expected

Reply:
Good idea, thanks =)